src/hooks/useGtm.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react'
+import { gtmClear, gtmInit, setChainId } from '@/services/analytics/gtm'
+import { useAppSelector } from '@/store'
+import { CookieType, selectCookies } from '@/store/cookiesSlice'
+import useChainId from './useChainId'
+
+const useGtm = () => {
+  const chainId = useChainId()
+  const cookies = useAppSelector(selectCookies)
+  const isAnalyticsEnabled = cookies[CookieType.ANALYTICS]
+
+  useEffect(() => {
+    isAnalyticsEnabled ? gtmInit() : gtmClear()
```

Comment:
I think this should go in the `cookiesMiddleware`.

Reply:
I think the opposite. Cookies is just a storage, it shouldn't control what modules are loaded. I would in fact move the Beamer init out of it.

Reply:
[Middleware are intended to contain logic with side effects.](https://redux.js.org/tutorials/fundamentals/part-4-store#middleware-use-cases)

Reply:
Related to the store, sure, but tracking and cookies are completely unrelated pieces of functionality. The only connection is that if the cookies aren't allowed, we shouldn't load the analytics. It doesn't mean the cookie storage should be responsible for loading GTM.

Reply:
By your logic, we shouldn't use modules and write the whole app in the same file.